libexec/scoop-update.ps1

```diff
@@ -54,16 +54,28 @@ if(($PSVersionTable.PSVersion.Major) -lt 5) {
     Write-Output "Upgrade PowerShell: https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-windows"
     break
 }
+$show_update_log = get_config 'show_update_log' $true
 
-function update_scoop() {
+function update_scoop($show_update_log) {
     # check for git
-    if(!(Test-CommandAvailable git)) { abort "Scoop uses Git to update itself. Run 'scoop install git' and try again." }
+    if (!(Test-CommandAvailable git)) { abort "Scoop uses Git to update itself. Run 'scoop install git' and try again." }
+
+    try {
+        $now = [System.DateTime]::Now.ToString('o')
+        $update_until = [System.DateTime]::Parse((get_config update_until $now))
+        if ((New-TimeSpan $update_until $now).TotalSeconds -lt 0) {
```

Comment:
```suggestion
        $update_until = [System.DateTime]::Parse((get_config update_until))
        if ((New-TimeSpan $update_until).TotalSeconds -lt 0) {
```

Why use `$now` as default? `get_config update_until` is okay since there's `try-catch`

Reply:
`[System.DateTime]::Parse` can not parser `$null`. If the user do not use `scoop hold scoop`, the `update_until` will not to be set and `get_config update_until` will return `$null`. That will be caught but should be silent.

Reply:
Hmm, got it, so I preferred using `if-else` in such case to make it clear.

Reply:
IF `update_until` is set, check it, IF NOT, go ahead.